SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -2151,212 +2151,129 @@ void CNEMOEulerSolver::BC_Outlet(CGeometry *geometry, CSolver **solver_container
-//
-//        /*--- Laminar viscosity ---*/
-//        visc_numerics->SetLaminarViscosity(nodes->GetLaminarViscosity(iPoint), nodes->GetLaminarViscosity(iPoint));
-//
-//        /*--- Compute and update residual ---*/
-//        auto residual = visc_numerics->ComputeResidual(config);
-//        LinSysRes.SubtractBlock(iPoint, residual);
-//
-//        /*--- Jacobian contribution for implicit integration ---*/
-//        //if (implicit)
-//        //  Jacobian.SubtractBlock2Diag(iPoint, residual.Jacobian_i);
-//      }
-//
-//    }
-//  }
-//
-//  /*--- Free locally allocated memory ---*/
-//  delete [] U_domain;
-//  delete [] U_inlet;
-//  delete [] V_domain;
-//  delete [] V_inlet;
-//  delete [] Normal;
+ /*--- Supersonic inlet flow: there are no outgoing characteristics,
+  so all flow variables can be imposed at the inlet.
+  First, retrieve the specified values for the primitive variables. ---*/
 
+ const string Marker_Tag = config->GetMarker_All_TagBound(val_marker);
+
+ const su2double Temperature = config->GetInlet_Temperature(Marker_Tag);
+ const su2double Pressure = config->GetInlet_Pressure(Marker_Tag);
+ const su2double* Velocity = config->GetInlet_Velocity(Marker_Tag);
+ const su2double* Mass_Frac = config->GetInlet_MassFrac(); 
+ su2double Temperature_ve = config->GetInlet_Temperature_ve();
+
+ /*--- If no Tve value specified (left as 0 K default), set to Ttr value ---*/
+ if (Temperature_ve == 0.0) {Temperature_ve = Temperature;}
+
+ /*--- Set mixture state ---*/
+ FluidModel->SetTDStatePTTv(Pressure, Mass_Frac, Temperature, Temperature_ve);
+
+ /*--- Compute Ma vector for flow direction ---*/
+ const su2double soundspeed = FluidModel->ComputeSoundSpeed();
+
+ su2double Mvec[MAXNDIM] = {0.0};
+
+ for (unsigned short iDim = 0; iDim < nDim; iDim++)
+   Mvec[iDim] = Velocity[iDim] / soundspeed;
+
+ /*--- Allocate inlet node to compute gradients for numerics ---*/
+ CNEMOEulerVariable node_inlet(Pressure, Mass_Frac, Mvec,
+                                        Temperature, Temperature_ve,
+                                        1, nDim, nVar, nPrimVar, nPrimVarGrad,
+                                        config, FluidModel);
+
+ node_inlet.SetPrimVar(0, FluidModel);
+ 
+ su2double Normal[MAXNDIM] = {0.0};
+ 
+ /*--- Loop over all the vertices on this boundary marker ---*/
+ for(unsigned long iVertex = 0; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+   const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+   /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
+   if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+   /*--- Index of the closest interior node ---*/
+   const auto Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+   /*--- Normal vector for this vertex (negate for outward convention) ---*/
+   geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
+   for (unsigned short iDim = 0; iDim < nDim; iDim++) Normal[iDim] = -Normal[iDim];
+
```

Comment:
@pcarruscag if we add the CharacPrimVar, would the below be alright?
`    auto* V_inlet = GetCharacPrimVar(val_marker, iVertex);
     V_inlet=node_inlet.GetPrimitive(0);
`

Reply:
No, that would change the pointer but not the data in CharacPrimVar, turbulence solvers downstream would not be reading the data used by the flow solver, which is the objective of CharacPrimVar